### PR TITLE
[DO-1618] Add text filter field for the PEP dataset && [DO-1610] Add …

### PR DESCRIPTION
**Pull request by AlenaYanish**
…number filter field for ID column in the PEP dataset && [DO-1947] Display the dates in PEP dataset to one view

### CODE REVIEW

File: src/components/filter-fields/FilterField.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      );
+    default:
+      return null;
+  }
+};
+
+FilterField.propTypes = {
+  filter: PropTypes.shape({
+    queryParam: PropTypes.string,
+    type: PropTypes.string,
```
Comment (roman-tiukh) [inline src/components/filter-fields/FilterField.jsx:66]: required

File: src/components/filter-fields/FilterField.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        <>
+          <div>
+            <SearchBox
+              className="border-gray-300 w-20"
+              size="md"
+              type="number"
+              placeholder="12345"
+              value={value}
+              onChange={(e) => onChange(filter.queryParam, e.target.value)}
+              onClear={() => onClear(filter.queryParam)}
```
Comment (roman-tiukh) [inline src/components/filter-fields/FilterField.jsx:35]: you can create onClean in  FilterField component

File: src/components/table/Table.jsx [hunk 3]
```diff
@@ -135,6 +190,10 @@ Table.propTypes = {
     prop: PropTypes.string.isRequired,
     width: PropTypes.string.isRequired,
     noSort: PropTypes.bool,
+    filter: PropTypes.shape({
+      queryParam: PropTypes.string,
+      type: PropTypes.string,
```
Comment (roman-tiukh) [inline src/components/table/Table.jsx:195]: required

File: src/components/table/Table.jsx [hunk 4]
```diff
@@ -102,9 +145,9 @@ const Table = (props) => {
                 <th
                   style={{ width: col.width }}
                   key={col.prop}
-                  className={`border-b-2 whitespace-no-wrap ${!col.noSort ? 'cursor-pointer' : ''}`}
+                  className="border-b-2 whitespace-no-wrap"
                 >
-                  <div className="flex items-center justify-between" onClick={() => handleHeaderClick(col)}>
+                  <div className={`flex items-center h-8 justify-between ${!col.noSort ? 'cursor-pointer' : ''}`} onClick={() => handleHeaderClick(col)}>
```
Comment (roman-tiukh) [inline src/components/table/Table.jsx:150]: too long line

**Review by roman-tiukh — CHANGES_REQUESTED** [review]
need a call)

**Review by roman-tiukh — APPROVED** [review]

**Review by EmmaSamoilova — APPROVED** [review]

**Pull request merged**